sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -139,6 +139,25 @@ def done(self):
       self._state = self.DONE_STATE
 
 
+class _PartitionBuffer(object):
+  """This class is created to support partition(n) function."""
+  def __init__(self, inputs):
+    self._inputs = inputs
+
+  def partition(self, n):
+    v = list(self._inputs.values())[0]
+    if isinstance(v, list):
+      return [self._inputs]
```

Comment:
We will also want to split this case up.

Reply:
I tried to split it, however, `test_pardo_timers` and other tests using `_run_pardo_state_timers()` cannot pass. A pattern I observed is when we have a timer, it has two elements at inputs and should be passed together, otherwise, the pipeline is hanging. How do we work around these cases?

an input example with a timer:
```
[{'Create/Read/Impulse': [b'\x7f\xdf;dZ\x1c\xac\t\x00\x00\x00\x01\x0f\x00'], 'ParDo(TimerDoFn)_timers_to_read_timer/Read': []}]
```
This will split into 
```
[{'Create/Read/Impulse': [b'\x7f\xdf;dZ\x1c\xac\t\x00\x00\x00\x01\x0f\x00']}, {'ParDo(TimerDoFn)_timers_to_read_timer/Read': []}]
```

Reply:
It turns out that we cannot split inputs with a timer. This is fixed at new commit.